### DISCUSSION TITLE
code lens gone and cannot run/debug functions via UI

### DISCUSSION THREAD

**Discussion by compiledsound** [golang/vscode-go#2064] [Help] [answered]
Checking configured tools....
GOBIN: undefined
toolsGopath: 
gopath: /Users/developer/go
GOROOT: /usr/local/Cellar/go/1.17.6/libexec
PATH: /Users/developer/.sdkman/candidates/java/current/bin:/Users/developer/.vulkan_sdk/macOS/bin:/Users/developer/.vulkan_sdk/macOS/bin:/Users/developer/.cargo/bin:/anaconda3/bin:/Users/developer/Development/flutter/bin:/Library/Frameworks/Python.framework/Versions/3.6/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Applications/Postgres.app/Contents/Versions/latest/bin:/Users/developer/go/bin

	go:	/usr/local/bin/go: go version go1.17.6 darwin/amd64

	gopkgs:	/Users/developer/go/bin/gopkgs: go1.17.6
		path	github.com/uudashr/gopkgs/v2/cmd/gopkgs
		mod	github.com/uudashr/gopkgs/v2	v2.1.2	h1:A0+QH6wqNRHORJnxmqfeuBEsK4nYQ7pgcOHhqpqcrpo=
		dep	github.com/karrick/godirwalk	v1.12.0	h1:nkS4xxsjiZMvVlazd0mFyiwD4BR9f3m6LXGhM2TUx3Y=
		dep	github.com/pkg/errors	v0.8.1	h1:iURUrRGxPUNPdy5/HRSm+Yj6okJ6UtLINN0Q9M4+h3I=
	
	go-outline:	/Users/developer/go/bin/go-outline: go version -m failed: Error: failed to run 'go version -m /Users/developer/go/bin/go-outline': /Users/developer/go/bin/go-outline: go version not found

	gotests:	/Users/developer/go/bin/gotests: go1.17.6
		path	github.com/cweill/gotests/gotests
		mod	github.com/cweill/gotests	v1.6.0	h1:KJx+/p4EweijYzqPb4Y/8umDCip1Cv6hEVyOx0mE9W8=
		dep	golang.org/x/tools	v0.0.0-20191109212701-97ad0ed33101	h1:LCmXVkvpQCDj724eX6irUTPCJP5GelFHxqGSWL2D1R0=
	
	gomodifytags:	not installed
	impl:	/Users/developer/go/bin/impl: go1.17.6
		path	github.com/josharian/impl
		mod	github.com/josharian/impl	v1.1.0	h1:gafhg1OFVMq46ifdkBa8wp4hlGogjktjjA5h/2j4+2k=
		dep	golang.org/x/mod	v0.2.0	h1:KU7oHjnv3XNWfa5COkzUifxZmxp1TyI7ImMXqFxLwvQ=
		dep	golang.org/x/tools	v0.0.0-20200522201501-cb1345f3a375	h1:SjQ2+AKWgZLc1xej6WSzL+Dfs5Uyd5xcZH1mGC411IA=
		dep	golang.org/x/xerrors	v0.0.0-20191204190536-9bdfabe68543	h1:E7g+9GITq07hpfrRu66IVDexMakfv52eLZ2CXBWiKr4=
	
	goplay:	not installed
	dlv:	/Users/developer/go/bin/dlv: go1.17.6
		path	github.com/go-delve/delve/cmd/dlv
		mod	github.com/go-delve/delve	v1.8.1	h1:X8OujRtMxTzNxM0y3Db+NEafbqEm1r52OydjDHZsrfs=
		dep	github.com/cilium/ebpf	v0.7.0	h1:1k/q3ATgxSXRdrmPfH8d7YK0GfqVsEKZAX9dQZvs56k=
		dep	github.com/cosiner/argv	v0.1.0	h1:BVDiEL32lwHukgJKP87btEPenzrrHUjajs/8yzaqcXg=
		dep	github.com/cpuguy83/go-md2man/v2	v2.0.0	h1:EoUDS0afbrsXAZ9YQ9jdu/mZ2sXgT1/2yyNng4PGlyM=
		dep	github.com/derekparker/trie	v0.0.0-20200317170641-1fdf38b7b0e9	h1:G765iDCq7bP5opdrPkXk+4V3yfkgV9iGFuheWZ/X/zY=
		dep	github.com/google/go-dap	v0.6.0	h1:Y1RHGUtv3R8y6sXq2dtGRMYrFB2hSqyFVws7jucrzX4=
		dep	github.com/hashicorp/golang-lru	v0.5.4	h1:YDjusn29QI/Das2iO9M0BHnIbxPeyuCHsjMW+lJfyTc=
		dep	github.com/mattn/go-isatty	v0.0.3	h1:ns/ykhmWi7G9O+8a448SecJU3nSMBXJfqQkl0upE1jI=
		dep	github.com/mattn/go-runewidth	v0.0.3	h1:a+kO+98RDGEfo6asOGMmpodZq4FNtnGP54yps8BzLR4=
		dep	github.com/peterh/liner	v1.2.2	h1:aJ4AOodmL+JxOZZEL2u9iJf8omNRpqHc/EbrK+3mAXw=
		dep	github.com/russross/blackfriday/v2	v2.0.1	h1:lPqVAte+HuHNfhJ/0LC98ESWRz8afy9tM/0RK8m9o+Q=
		dep	github.com/shurcooL/sanitized_anchor_name	v1.0.0	h1:PdmoCO6wvbs+7yrJyMORt4/BmY5IYyJwS/kOiWx8mHo=
		dep	github.com/sirupsen/logrus	v1.6.0	h1:UBcNElsrwanuuMsnGSlYmtmgbb23qDR5dG+6X6Oo89I=
		dep	github.com/spf13/cobra	v1.1.3	h1:xghbfqPkxzxP3C/f3n5DdpAbdKLj4ZE4BWQI362l53M=
		dep	github.com/spf13/pflag	v1.0.5	h1:iy+VFUOCP1a+8yFto/drg2CJ5u0yRoB7fZw3DKv/JXA=
		dep	go.starlark.net	v0.0.0-20200821142938-949cc6f4b097	h1:YiRMXXgG+Pg26t1fjq+iAjaauKWMC9cmGFrtOEuwDDg=
		dep	golang.org/x/arch	v0.0.0-20190927153633-4e8777c89be4	h1:QlVATYS7JBoZMVaf+cNjb90WD/beKVHnIxFKT4QaHVI=
		dep	golang.org/x/sys	v0.0.0-20211117180635-dee7805ff2e1	h1:kwrAHlwJ0DUBZwQ238v+Uod/3eZ8B2K5rYsUHBQvzmI=
		dep	gopkg.in/yaml.v2	v2.4.0	h1:D8xgwECY7CYvx+Y2n4sBz93Jn9JRvxdiyyo8CTfuKaY=
	
	staticcheck:	/Users/developer/go/bin/staticcheck: go1.17.6
		path	honnef.co/go/tools/cmd/staticcheck
		mod	honnef.co/go/tools	v0.2.2	h1:MNh1AVMyVX23VUHE2O27jm6lNj3vjO5DexS4A1xvnzk=
		dep	github.com/BurntSushi/toml	v0.3.1	h1:WXkYYl6Yr3qBf1K79EBnL4mak0OimBfB0XUf9Vl28OQ=
		dep	golang.org/x/mod	v0.3.0	h1:RM4zey1++hCTbCVQfnWeKs9/IEsaBLA8vTkd0WVtmH4=
		dep	golang.org/x/sys	v0.0.0-20210119212857-b64e53b001e4	h1:myAQVi0cGEoqQVR5POX+8RR2mrocKqNN1hmeMqhX27k=
		dep	golang.org/x/tools	v0.1.0	h1:po9/4sTYwZU9lPhi1tOrb4hCv3qrhiQ77LZfGa2OjwY=
		dep	golang.org/x/xerrors	v0.0.0-20200804184101-5ec99f83aff1	h1:go1bK/D/BFZV2I8cIQd1NKEZ+0owSTG1fDTci4IqFcE=
	
	gopls:	/Users/developer/go/bin/gopls: go1.17.6
		path	golang.org/x/tools/gopls
		mod	golang.org/x/tools/gopls	v0.7.5	h1:8Az52YwcFXTWPvrRomns1C0N+zlgTyyPKWvRazO9GG8=
		dep	github.com/BurntSushi/toml	v0.4.1	h1:GaI7EiDXDRfa8VshkTj7Fym7ha+y8/XxIgD2okUIjLw=
		dep	github.com/google/go-cmp	v0.5.6	h1:BKbKCqvP6I+rmFHt06ZmyQtvB8xAkWdhFyr0ZUNZcxQ=
		dep	github.com/sergi/go-diff	v1.1.0	h1:we8PVUC3FE2uYfodKH/nBHMSetSfHDR6scGdBi+erh0=
		dep	golang.org/x/mod	v0.5.1	h1:OJxoQ/rynoF0dcCdI7cLPktw/hR2cueqYfjm43oqK38=
		dep	golang.org/x/sync	v0.0.0-20210220032951-036812b2e83c	h1:5KslGYwFpkhGh+Q16bwMP3cOontH8FOep7tGV86Y7SQ=
		dep	golang.org/x/sys	v0.0.0-20211019181941-9d821ace8654	h1:id054HUawV2/6IGm2IV8KZQjqtwAOo2CYlOToYqa0d0=
		dep	golang.org/x/text	v0.3.7	h1:olpwvP2KacW1ZWvsR7uQhoyTYvKAupfQrRGBFM352Gk=
		dep	golang.org/x/tools	v0.1.9-0.20220114220130-fd7798718afd	h1:lTnuArxJC+n54TyvWUPyHhrnGxYvhSi13/aM2Ndr4bs=
		dep	golang.org/x/xerrors	v0.0.0-20200804184101-5ec99f83aff1	h1:go1bK/D/BFZV2I8cIQd1NKEZ+0owSTG1fDTci4IqFcE=
		dep	honnef.co/go/tools	v0.2.1	h1:/EPr//+UMMXwMTkXvCCoaJDq8cpjMO80Ou+L4PDo2mY=
		dep	mvdan.cc/gofumpt	v0.1.1	h1:bi/1aS/5W00E2ny5q65w9SnKpWEF/UIOqDYBILpo9rA=
		dep	mvdan.cc/xurls/v2	v2.3.0	h1:59Olnbt67UKpxF1EwVBopJvkSUBmgtb468E4GVWIZ1I=
	

go env
Workspace Folder (server): /Users/developer/Development/server
	GO111MODULE=""
	GOARCH="amd64"
	GOBIN=""
	GOCACHE="/Users/developer/Library/Caches/go-build"
	GOENV="/Users/developer/Library/Application Support/go/env"
	GOEXE=""
	GOEXPERIMENT=""
	GOFLAGS=""
	GOHOSTARCH="amd64"
	GOHOSTOS="darwin"
	GOINSECURE=""
	GOMODCACHE="/Users/developer/go/pkg/mod"
	GONOPROXY=""
	GONOSUMDB=""
	GOOS="darwin"
	GOPATH="/Users/developer/go"
	GOPRIVATE=""
	GOPROXY="https://proxy.golang.org,direct"
	GOROOT="/usr/local/Cellar/go/1.17.6/libexec"
	GOSUMDB="sum.golang.org"
	GOTMPDIR=""
	GOTOOLDIR="/usr/local/Cellar/go/1.17.6/libexec/pkg/tool/darwin_amd64"
	GOVCS=""
	GOVERSION="go1.17.6"
	GCCGO="gccgo"
	AR="ar"
	CC="clang"
	CXX="clang++"
	CGO_ENABLED="1"
	GOMOD="/Users/developer/Development/server/go.mod"
	CGO_CFLAGS="-g -O2"
	CGO_CPPFLAGS=""
	CGO_CXXFLAGS="-g -O2"
	CGO_FFLAGS="-g -O2"
	CGO_LDFLAGS="-g -O2"
	PKG_CONFIG="pkg-config"
	GOGCCFLAGS="-fPIC -arch x86_64 -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/1x/lvsm8tjx3fl_283vktyjhlz80000gn/T/go-build4225792037=/tmp/go-build -gno-record-gcc-switches -fno-common"
Go Extension version: v0.31.1

I recently updated to Monterey 12.2.1 and noticed that the "run test | debug test" buttons disappeared. The Test Explorer seems to be having issues as well. Every time I click on a dropdown, the label disappears. Im also not able to use the debug/test Function at cursor command.

I removed vscode and the supporting files and tried reinstalling with no luck.

I can test/debug functions is via command line or by creating a launch configuration that takes a prompt.
i.e.
```json
"configurations": [
        {
            "name": "Debug test",
            "type": "go",
            "request": "launch",
            "mode": "test",
            "program": "${fileDirname}",
            "args": [
                "-test.run",
                "${input:testName}"
            ]
        }
    ],
    "inputs": [
        {
            "type": "promptString",
            "id": "testName",
            "description": "Enter Test Name",
            "default": "TestName"
        }
      ]
```
But would like to get the UI working like it use to. Has anyone had this issue?

**Comment by hyangah** [comment, marked as answer]
Can you try to reinstall the `go-outline` tool and see if that helps? 
"Go: Install/Update Tools" -> select "go-outline".
After tool update is successful,  "Developers: Reload WIndow" to start from clean state.

**Reply by compiledsound** [reply]
perfect. curious how that fixed it, since vs code was working fine until I updated macOS (which may not be the reason)

**Reply by hyangah** [reply]
Great. 
`go-outline` is the tool that provides symbol information test codelens and test explorer depend on. 
https://github.com/golang/vscode-go/blob/master/docs/tools.md#go-outline

I noticed 
```
go-outline:	/Users/developer/go/bin/go-outline: go version -m failed: Error: failed to run 'go version -m /Users/developer/go/bin/go-outline': /Users/developer/go/bin/go-outline: go version not found
``` 
from your report.  That indicates something is wrong with the `go-outline` binary.  

Another news: from the next release, `go-outline` will retire and the extension will let `gopls` provide the symbol information.  One less binary to depend on. :-)

**Reply by changliu-wk** [reply]
I tried to individually install go-outline, but I got this error:
go-outline: failed to install go-outline(github.com/ramya-rao-a/go-outline@latest): Error: Command failed: /usr/local/bin/go install -v github.com/ramya-rao-a/go-outline@latest
go install: github.com/ramya-rao-a/go-outline@latest: no matching versions for query "latest"

**Reply by hyangah** [reply]
@changliu-wk Try `go install -x github.com/ramya-rao-a/go-outline@latest` - that will tell you what claims there is no matching version for `latest`. (Likely a bad proxy).

**Reply by changliu-wk** [reply]
> go install -x github.com/ramya-rao-a/go-outline@latest

Yes, it is the bad proxy. Now I am good, thanks a lot! Saved my day!
